frontend/src/core/utils/fluent/flattenElements.js

```diff
@@ -8,25 +8,45 @@ import type { FluentElement } from './types';
 /**
  * Return a flattened list of Fluent elements.
  *
- * Takes a list of Fluent SyntaxNode elements, extracts a serialized value for
- * each and returns an array with a single TextElement containing those values.
+ * @param {Array<SyntaxNode>} elements A list of Fluent SyntaxNode elements to flatten.
  *
- * @param {Array<SyntaxNode>} elements A list of Fluent SyntaxNode lements to flatten.
- *
- * @returns {Array<TextElement>} An array containing a single TextElement which
- * contains all elements' values serialized.
+ * @returns {Array<TextElement|Placeable>} An array containing elements of type
+ * TextElement (merging serialized values of neighbour simple elements) and
+ * Placeable (representing select expressions).
  */
 export default function flattenElements(elements: Array<FluentElement>): Array<FluentElement> {
-    const values = elements.map(element => {
-        switch (element.type) {
-            case 'TextElement':
-                return element.value;
-            default:
-                return serializeExpression(element);
+    const flatElements = [];
+    let simpleElements = [];
```

Comment:
As I want changes here below, how about `textFragments`? They're not simple nor elements in the end, right?